# Introduction

The data comes from an experiment performed in Cotney lab 4 years ago. I have retrieved the fastq files in the following directory : `/archive/labs/Cotney/DATA/STARR-Seq/ChIPSTARRseq_test/test_run_3/TYAN01*`.

* Link to the hg19 reference genome : https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips. I downloaded the **hg19.fa.gz** file
* Link to the list of the chrommosome size : https://gist.github.com/nimezhu/d97a302816ea751e25571773d075b35e

# Description of the analysis steps

## Step01 : Organization of fastq files
* For this, I used the script named :

## Step02 : Mapping the reads
* For this, I used the script named :

## Step03 : Filtering BAM files
* For this, I used the script named :

## Step04 : Indexing BAM files
* For this, I used the script named :

## Step05 : Extract fragments
* For this, I used the script named :

## Step06 : Peak calling
* For this, I used the script named :
